007-OFEP-flag-change-events.md

```diff
@@ -0,0 +1,91 @@
+## 007-OFEP-flag-change-events
+
+## State: DRAFTING
+
+Some flag SDKs support listening for flag value changes or general configuration changes ([launchdarkly](https://docs.launchdarkly.com/sdk/features/flag-changes), [cloudbees](https://docs.cloudbees.com/docs/cloudbees-feature-management/latest/reporting/configuration-fetched-handler), flagd). This can allow us to use an event-based paradigm for consuming flags. Client apps may use feature flags for characteristics that aren't specifically tied to a user-action, making imperative flag evaluation a less-than-ideal solution. Web clients using frameworks such as React need a way to register a callback for when a provider has initialized, given that it's not uncommon for a flag to be evaluated before the provider is ready (during page load). A server application could listen to a flag that changes some operational behavior.
+
+Examples:
+
+- A web client's banner color could be updated when an associated flag is changed, without any user action or a page reload.
+- A service could subscribe to a flag that controls its global log-level.
+- A subsystem could be restarted with the new flag value.
+
+## Design
+
+The provider interface and the OpenFeature client would be extended to have new functionality to register handlers for a set of events defined by the SDK (`ProviderEvents`). When a consumer (_application integrator_, _application author_, _integration author_) registers a handler on a client, the client maintains this handler. The provider emits events or runs a callback indicating that it received a certain event, optionally providing data associated with that event. The callbacks registered with the client are then invoked with this data (`EventData`).
```

Comment:
>the client maintains this handler

"client maintains" or "api maintains"? Which level of the hierarchy do these sit?

Reply:
I've proposed (and PoC'd) this functionality exists on the client.